packages/modal/src/Modal.tsx

```diff
@@ -0,0 +1,259 @@
+import React, { useCallback, SetStateAction, useRef } from 'react';
+import PropTypes from 'prop-types';
+import Portal from '@leafygreen-ui/portal';
+import { Global } from '@emotion/core';
+import Icon, { Size } from '@leafygreen-ui/icon';
+import { useEventListener } from '@leafygreen-ui/hooks';
+import { uiColors } from '@leafygreen-ui/palette';
+import { css, cx } from '@leafygreen-ui/emotion';
+
+export const ModalSize = {
+  XXSmall: 'xxsmall',
+  XSmall: 'xsmall',
+  Small: 'small',
+  Default: 'default',
+  Large: 'large',
+  XLarge: 'xlarge',
+} as const;
+
+export type ModalSize = typeof ModalSize[keyof typeof ModalSize];
+
+const defaultSpacing = 18;
+
+const scrollContainer = css`
+  animation: fade-in 500ms cubic-bezier(0.165, 0.84, 0.44, 1);
+  -webkit-animation: fade-in 500ms cubic-bezier(0.165, 0.84, 0.44, 1);
+  background-color: ${uiColors.black};
+  opacity: 0.6;
+  overflow-y: auto;
+  position: fixed;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  display: flex;
+  justify-content: center;
+  align-items: flex-start;
+  pointer-events: initial;
+`;
+
+const modalContentStyle = css`
+  margin: ${defaultSpacing}px auto;
+  max-width: 1270px;
+  min-width: 944px;
```

Comment:
This component....we may want to be responsive. So it can work on mobile devices. On mms we're using [facepaint](https://github.com/emotion-js/facepaint) to success.

Reply:
Is there a standard for the number of/places where we use breakpoints?

Reply:
Happy to come up with some for this use case, but was wondering if there were certain breakpoints that we use as benchmarks and then back into styles from there?

Reply:
Added breakpoints based on the small, medium and large screen sizes defined in MMS. The only size that was effected was the large size. Let me know what you think @hswolff!